### PR TITLE
[SW-2437] relocate to ai.h2o package

### PR DESCRIPTION
**Pull request by honzasterba**

### CODE REVIEW

File: jvm-packages/xgboost4j/src/native/xgboost4j.cpp [hunk 1]
```diff
@@ -344,6 +344,33 @@ JNIEXPORT jint JNICALL Java_ml_dmlc_xgboost4j_java_XGBoostJNI_XGDMatrixCreateFro
   return ret;
 }
 
+/*
+ * Class:     ml_dmlc_xgboost4j_java_XGBoostJNI
+ * Method:    XGDMatrixCreateFrom2DCSCEx
+ * Signature: ([J[J[F)J
+ */
+JNIEXPORT jint JNICALL Java_ml_dmlc_xgboost4j_java_XGBoostJNI_XGDMatrixCreateFrom2DCSCEx
```
Comment (mn-mikke) [inline jvm-packages/xgboost4j/src/native/xgboost4j.cpp:352]: Why this extra method is needed?

Comment (honzasterba) [reply jvm-packages/xgboost4j/src/native/xgboost4j.cpp:352]: this is resurected method that was mistankenly deleted in the past

Comment (mn-mikke) [reply jvm-packages/xgboost4j/src/native/xgboost4j.cpp:352]: Ok, thanks!

**Review by mn-mikke — APPROVED** [review]
Code as such looks good to me! As we discussed privately, still want to understand if we can reference two different version of a native library from one JVM.

**Comment by michalkurka** [comment]
I like this approach!

An alternative approach would be to use the existing java sources and "generate" new sources with packages relocated.

This would allow us to publish also source code (jars) that is corresponding exactly to the artifacts.

    ml.dmlc.xgboost4j.java -> ai.h2o.xgboost4j.java
Since we are doing a straightforward move a simple (recursive) `sed` would do the job.



WDYT?

**Review by michalkurka — APPROVED** [review]

**Comment by honzasterba** [comment]
@mn-mikke I did some research and loading two libs with same symbols should not be an issue as long as they are under different classloaders

**Comment by mn-mikke** [comment]
@honzasterba Perfect! Thanks for checking it.

**Pull request merged**